packages/chat-headless/src/ChatHeadlessImpl.ts

```diff
@@ -264,6 +264,21 @@ export class ChatHeadlessImpl implements ChatHeadless {
   }
 
   restartConversation() {
+    if (isChatEventClient(this.chatClient)) {
+      this.chatClient.resetSession();
+      let nextClient: ChatClient | undefined = undefined;
```

Comment:
I think we should always set to bot client when restarting convo, instead of next client